Edits to enhance readability of Run a Local Testnet tutorial doc (#687)
aptos-labs/aptos-core: developer-docs-site/docs/tutorials/run-a-local-testnet.md

# Run a Local Testnet

You can run a local testnet of the Aptos Blockchain. This local testnet will not be connected to the Aptos devnet. It will run on your local machine, independently of the Aptos ecosystem. You can use this local testnet for testing and development purposes.

You can run a local testnet in two ways:

1. Using the Aptos-core source code. This approach is useful for testing modifications to the Aptos-core codebase or to the Aptos Framework.

2. Using Docker. This is particularly useful for building services on top of the Aptos Blockchain or the Aptos Framework, as there is no build overhead and the ledger persists across network restarts (by default).

The rest of this document describes:

- How to start your local testnet using both the methods, and
- How to start a Faucet service and attach it to your testnet.

## Using the Aptos-core source code

1. Clone the Aptos-core repository from GitHub.

  ```
  git clone https://github.com/aptos-labs/aptos-core.git
  ```

> 

Author: I used the clone of my fork. Should we recommend fork plus clone, instead of cloning Aptos-core directly?

Reviewer: yes, that's ideal.

Author: Done.

# Run a Local Testnet

> You can run a local testnet of the Aptos Blockchain. This local testnet will not be connected to the Aptos devnet. It will run on your local machine, independently of the Aptos ecosystem. You can use this local testnet for testing and development purposes.

Reviewer: Maybe tweak the wording here...
```suggestion
You can run a local testnet of the Aptos Blockchain. This local testnet will not be connected to the Aptos devnet. It will run on your local machine, independent of other Aptos networks. You can use this local testnet for testing and development purposes.
```

    ```
    git clone https://github.com/<YOUR-GITHUB-USERID>/aptos-core

    ```

2. `cd` into `aptos-core` directory.

    ```
    cd aptos-core
    ```

3. Run the `scripts/dev_setup.sh` Bash script as shown below. This will prepare your developer environment.

    ```
    ./scripts/dev_setup.sh
    ```

4. Update your current shell environment.

    ```
    source ~/.cargo/env
    ```

5. With your development environment ready, now you can start your testnet network. Before you proceed, make a note of the following:

  - When you run the below command to start the local testnet, your terminal will enter into an interactive mode, with an option to terminate the testnet. Hence, you will need to open another shell terminal for the subsequent steps described in this section.
  - After the below command runs, you will need to copy the `Config path` information from the terminal output for the next step.

    To start your testnet locally, run the following command:

    ```
    CARGO_NET_GIT_FETCH_WITH_CLI=true cargo run -p aptos-node -- --test
    ```

    See below for an example of the partial output. Make a note of the `Config path` from the output.

    ```
    ...

>     ...

Author: Looks fine in preview.

The rest of this document describes:

> - How to start your local testnet with a single Validator FullNode, using both the methods, and

Author: @davidiw Is this accurate?

      --server-url http://127.0.0.1:8080
```

> This will start a Faucet running locally without any restrictions to tokens that can be claimed and minted. This Faucet service will be as accessible as the testnet youy started above.

Reviewer: ```suggestion
This will start a Faucet running locally without any restrictions to tokens that can be claimed and minted. This Faucet service will be as accessible as the testnet you started above.
```

## Using Docker

This section describes how to start your local testing using Docker.

1. Install [Docker](https://docs.docker.com/get-docker/) including [Docker-Compose](https://docs.docker.com/compose/install/).
2. Create a directory for your local test validator network, and `cd` into it.

> 3. Download the YAML files for:

Reviewer: ```suggestion
3. Download the YAML configuration files for:
```

  - [Validator testnet docker compose](https://github.com/aptos-labs/aptos-core/blob/main/docker/compose/validator-testnet/docker-compose.yaml) and
  - [Validator configuration](https://github.com/aptos-labs/aptos-core/blob/main/docker/compose/validator-testnet/validator_node_template.yaml).

4. Start Docker Compose by running the command:

    ```
    docker-compose up
    ```

### Example

For example, the command sequence for the above steps 2 through 4 is shown below:

  **NOTE**: If you do not have `wget` installed on your macOS, then run the below command to install the `wget` before you follow the below example.

  ```
  brew install wget
  ```

  After you installed `wget` proceed as below.

> 

Reviewer: Maybe we should consider using tables if we want to get detailed with platform specific commands. I think someone reviewing this should know how to install wget, so I'd prefer to leave it off. Especially since this is a Mac centric approach to doing this.

If we do want to go down this approach, we could consider tables for linux, windows, mac: https://docusaurus.io/docs/markdown-features/tabs

Author: Removed it.

Aptos is running, press ctrl-c to exit
```

> Use the [Aptos CLI tool](https://github.com/aptos-labs/aptos-core/blob/main/crates/aptos/README.md) to interact with your local testnet. The above output contains information you will use for starting the [Aptos CLI tool](https://github.com/aptos-labs/aptos-core/blob/main/crates/aptos/README.md):

Author: This is accurate? I mean, the CLI tool is used to interact with the local testnet, I think?

* `Aptos root key path`: The root key (also known as the mint or faucet key) controls the account that can mint tokens. Available in the docker compose folder under `aptos_root_key`.
* `Waypoint`: A verifiable checkpoint of the blockchain (available in the docker compose folder under waypoint.txt)
* `REST endpoint`: The endpoint for the REST service, e.g., `http://127.0.0.1:8080`.

> * `ChainId`: The chain ID uniquely distinguishes this network from other blockchain networks.

Author: I believe the `chainID` is no longer required while using the CLI tool `aptos`, I think? @gregnazario?

## Next steps

At this point, you will have a special root account at `0x1` that can perform the mint operation. Follow up with:

* [Your first transaction](/tutorials/your-first-transaction) to learn how to submit transactions.
* [Your first Move module](/tutorials/your-first-move-module) to learn how to create Move modules.
* [Interacting with the Aptos Blockchain](/transactions/interacting-with-the-aptos-blockchain) to learn how to mint coins.

:::info

This guide does not describe how to create a faucet.

> 

Author: I am muddled on this statement. Both the methods actually start the faucet service, isn't it? Is this statement saying something different?
